main.go

```diff
@@ -97,12 +105,16 @@ func main() {
 		"use the demo provisioner to set host states")
 	flag.StringVar(&healthAddr, "health-addr", ":9440",
 		"The address the health endpoint binds to.")
+	flag.IntVar(&webhookPort, "webhook-port", 9443,
+		"Webhook Server port (set to 0 to disable)")
```

Comment:
I'm still continuing to find the decision to use the value `0` for a port to disable a whole new feature not a good interface design: it's not self-speaking (as the previously discussed `enable-webhook`), and could convey a misleading message (since it could be interpreted as _"choose the best port"_).
Configuration settings are effectively  part of the public API exposed by the component to the user, so having a dedicated setting to enable/disable a feature and another one to configure it (when enabled) could be clearer

Reply:
i think this is quite standard when using controller-runtime

Reply:
For enabling / disabling specifically webhooks? Kubebuilder [sample](https://book.kubebuilder.io/cronjob-tutorial/running.html?highlight=ENABLE_WEB#running-and-deploying-the-controller) uses an env var (which could be handy especially when testing without the webhook) whereas some operators could use a more explicit approach.

Anyhow, on a second thought, I'm wondering if this kind of configuration (the webhook port) it's really required or if it could be avoided at all.